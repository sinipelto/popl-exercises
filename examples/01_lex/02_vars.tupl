MEANING <- 42.
N <- 7.
variable <- 7 + 3.
<mytuple> <- [ 1 , 2, 1+2 ].
