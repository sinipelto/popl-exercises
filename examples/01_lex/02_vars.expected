LexToken(constIDENT,'MEANING',1,0)
LexToken(LARROW,'<-',1,8)
LexToken(NUMBER_LITERAL,42,1,11)
LexToken(DOT,'.',1,13)
LexToken(constIDENT,'N',2,15)
LexToken(LARROW,'<-',2,17)
LexToken(NUMBER_LITERAL,7,2,20)
LexToken(DOT,'.',2,21)
LexToken(varIDENT,'variable',3,23)
LexToken(LARROW,'<-',3,32)
LexToken(NUMBER_LITERAL,7,3,35)
LexToken(PLUS,'+',3,37)
LexToken(NUMBER_LITERAL,3,3,39)
LexToken(DOT,'.',3,40)
LexToken(tupleIDENT,'<mytuple>',4,42)
LexToken(LARROW,'<-',4,52)
LexToken(LSQUARE,'[',4,55)
LexToken(NUMBER_LITERAL,1,4,57)
LexToken(COMMA,',',4,59)
LexToken(NUMBER_LITERAL,2,4,61)
LexToken(COMMA,',',4,62)
LexToken(NUMBER_LITERAL,1,4,64)
LexToken(PLUS,'+',4,65)
LexToken(NUMBER_LITERAL,2,4,66)
LexToken(RSQUARE,']',4,68)
LexToken(DOT,'.',4,69)
